### PR TITLE
Feature/10 remove support for branches

### LINKED ISSUE #10: Remove support for branches
## Background
In the GH action there will not be support for branches check

## Feature
Remove all branches' logic.
Remove boolean outputs - action will fail with log message when a problem occurs.

### PR DESCRIPTION
**Pull request by miroslavpojer**
Release Notes:
- Removed all code part related to branches.
- Removed input: branch
- Removed input: fail-on-error
- Removed output: valid
- When problem found action fails.

Closes #10

### CODE REVIEW

File: version_tag_check/version_validator.py [hunk 1]
```diff
@@ -52,27 +52,56 @@ def __get_latest_version(self) -> Optional[Version]:
...
+        filtered_versions = self.__get_filtered_versions(nv.major, nv.minor)
+        if filtered_versions:
+            latest_filtered_version = max(filtered_versions)
+            logger.debug("Validator: Latest filtered version: %s", latest_filtered_version)
+
+            # Validate against the latest filtered version
+            if nv.major == latest_filtered_version.major and nv.minor == latest_filtered_version.minor:
+                if nv.patch == latest_filtered_version.patch + 1:
+                    return True
+                logger.error(f"New tag {nv} is not one patch higher than the latest tag {latest_filtered_version}.")
```
Comment (MobiTikula) [inline version_tag_check/version_validator.py:94]: dont use f strings for logging.

File: version_tag_check/version_validator.py [hunk 2]
```diff
@@ -52,27 +52,56 @@ def __get_latest_version(self) -> Optional[Version]:
...
+
+        # Check if this is a valid minor or major bump
+        if nv.major == latest_version.major:
+            if nv.minor == latest_version.minor + 1:
+                if nv.patch == 0:
+                    return True
+                logger.error(f"New tag {nv} is not a valid minor bump. Latest version: {latest_version}.")
```
Comment (MobiTikula) [inline version_tag_check/version_validator.py:101]: dont use f strings for logging.

File: version_tag_check/version_validator.py [hunk 3]
```diff
@@ -52,27 +52,56 @@ def __get_latest_version(self) -> Optional[Version]:
...
+        elif nv.major == latest_version.major + 1:
+            if nv.minor == 0 and nv.patch == 0:
+                return True
+            logger.error(f"New tag {nv} is not a valid major bump. Latest version: {latest_version}.")
```
Comment (MobiTikula) [inline version_tag_check/version_validator.py:105]: Same.

**Review by MobiTikula — COMMENTED** [review]
Please react to comments.

File: README.md [hunk 4]
```diff
@@ -73,10 +59,8 @@ See the default action step definition:
   env:
     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  
   with:
-    github-repository: "{ org }/{ repo }"
+    github-repository: "{ org }/{ repo }"   # ${{ github.repository }}
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply README.md]: Added e.g. in commit - e00dc5180fe130a878d25a8442c79c68bca6253e.

File: version_tag_check/version_validator.py (hunk 2 above)
Comment (miroslavpojer) [reply version_tag_check/version_validator.py]: Fixed in commit e00dc5180fe130a878d25a8442c79c68bca6253e.

**Pull request merged**